### ISSUE TITLE
Bug: *-type-checked-only configs are not exported

### ISSUE THREAD

**Issue by fpapado** [typescript-eslint/typescript-eslint#8599]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

The -type-checked-only configs seem to not be exposed, either in flat or legacy config. By using v7.1.1 (latest), I get an error when extending them.

I thought that they would be exposed, since they show up in the [changelog for 7.1.0](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v7.1.0). I might be missing something, for example whether it is intended 😅 

I will open a PR for this promptly, but thought to drop an issue in case I can provide any more information 😌 

![Terminal view of the flat config with -type-checked-only. A loading error is raised, around recommendedTypeCheckedOnly being undefined.](https://github.com/typescript-eslint/typescript-eslint/assets/3210764/796a4d28-2e19-4eb6-bb28-9107da0e5c0f)

![Terminal view of the legacy config with -type-checked-only. A loading error is raised, around recommended-type-checked-only not being found.](https://github.com/typescript-eslint/typescript-eslint/assets/3210764/b85cb2e7-1f74-406c-9fd7-f6c70c426e0b)

### Reproduction Repository Link

https://github.com/fpapado/typescript-eslint-type-checked-only-missing

### Repro Steps

1) clone the repo

2) Flat config:

```shell
cd flat-config
pnpm install
pnpm eslint .
```

3) Legacy config:

```shell
cd legacy-config
pnpm install
pnpm eslint .
```

An error should show up in both cases.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `7.1.0` |

**Comment by JoshuaKGoldberg** [comment]
Yeah, the configs not being exported was a consequence of their work happening around the same time as flat config work. @bradzacher did warn me something like this could happen!

Thanks for filing!
